packages/cli/src/resolvers/http-and-file.ts

```diff
@@ -0,0 +1,37 @@
+import { Resolver } from '@stoplight/json-ref-resolver';
```

Comment:
Is this a copy and paste from Spectral? Could we hoist this up somewhere and have it shared?

Reply:
This is a copy paste from spectral, that's correct. I think the `json-ref-resolver` could have such utility out of the box @P0lip @marbemac; would you be open to that?

Reply:
I don't think `json-ref-resolver` should provide out-of-the-box support for that.
That said, we could export such a resolver from `json-ref-resolver` pkg, but I'm not the owner of it, so it's not my call.

Reply:
> we could export such a resolver from json-ref-resolver

Yeah sorry that was what I meant. Precanned resolver.

Reply:
Right, making them available from the package is what we are advocating for here, not changing the default resolver. The goal is to move it somewhere accessible for both Spectral and Prism so we don’t have to copy and paste.

Reply:
`json-ref-resolver` sounds like the right place.
@XVincentX mind opening the PR?

Reply:
Hmm not too sure it's the right place, now that I think about it. Its job is really to resolve references — adding this capability would require to add an http client *and* the yaml parsing library to the project which I'd say clearly do not belong there. We should probably put this stuff somewhere else.

Reply:
It depends what we think resolving is. Resolving for many people is literally just the lookup, not the loader. JSON Ref Resolver right now is a resolve + file loader. 

File loading and URL loading is both loading, so maybe we should make a new package for some common loaders we have, and let people define their own? 

So they could use our resolver with their own loader, or their own resolver entirely... this is a bigger job but if we think that direction is viable, we can out this copy paste in now with a story on the backlog to pay off the tech debt later. 

Also interested in @marc and @P0lip opinions here

Reply:
> File loading and URL loading is both loading, so maybe we should make a new package for some common loaders we have, and let people define their own?

Correct, this is the direction I think we should go. Can we leave it as a next step?

Reply:
Hey @XVincentX I think it makes sense to extract this to its own package. Can you create an issue for that?

Reply:
We did it already, #493

Reply:
@XVincentX @philsturgeon 
> File loading and URL loading is both loading, so maybe we should make a new package for some common loaders we have, and let people define their own?

Yeah, a new package might be in fact a better choice.
Bear in mind that Spectral comes with its **own** `parseResolveResult` method, so it won't benefit as much as it might seem.
That said, I'm still leaning towards extracting those resolvers somewhere.